code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java

```diff
@@ -195,75 +221,51 @@ protected DeploymentUnitInstance createInstance(String launchConfigName, Service
     }
 
     @SuppressWarnings("unchecked")
-    protected List<Integer> getVolumesFromInstancesIds(Service service, String launchConfigName) {
-        List<Integer> volumesFromInstanceIds = new ArrayList<>();
-        Object volumesFromLaunchConfigs = ServiceDiscoveryUtil.getLaunchConfigObject(service, launchConfigName,
-                ServiceDiscoveryConstants.FIELD_DATA_VOLUMES_LAUNCH_CONFIG);
-        Object volumesFromInstance = ServiceDiscoveryUtil.getLaunchConfigObject(service, launchConfigName,
-                DockerInstanceConstants.FIELD_VOLUMES_FROM);
-        if (volumesFromInstance != null) {
-            volumesFromInstanceIds.addAll((List<Integer>) volumesFromInstance);
+    protected List<Integer> getSidekickContainersId(Service service, String launchConfigName, SidekickType sidekickType) {
+        List<Integer> sidekickInstanceIds = new ArrayList<>();
+        Object sidekickInstances = ServiceDiscoveryUtil.getLaunchConfigObject(service, launchConfigName,
+                sidekickType.launchConfigFieldName);
+        if (sidekickInstances != null) {
+            if (sidekickType.isList) {
+                sidekickInstanceIds.addAll((List<Integer>)sidekickInstances);
+            } else {
+                sidekickInstanceIds.add((Integer) sidekickInstances);
+            }
         }
-        
-        if (volumesFromLaunchConfigs != null) {
-            for (String volumesFromLaunchConfig : (List<String>) volumesFromLaunchConfigs) {
+
+        Object sidekicksLaunchConfigObj = ServiceDiscoveryUtil.getLaunchConfigObject(service, launchConfigName,
+                sidekickType.launchConfigType);
+        if (sidekicksLaunchConfigObj != null) {
+            List<String> sidekicksLaunchConfigIds = new ArrayList<>();
```

Comment:
Minor: The name 'sidekicksLaunchConfigIds' was a little confusing since the elements were compared against service.getName().  And I usually don't think of service name as an ID.